scoap3/cli_harvest.py

```diff
@@ -50,6 +54,31 @@ def get_packages_for_file_or_folder(source_file, source_folder):
     return entries
 
 
+def schedule_and_wait_crawl(*args, **kwargs):
+    """Calls inspire-crawler schedule_task and waits for the created task to finish."""
+
+    job_id = schedule_crawl(*args, **kwargs)
+    log('Crawler job scheduled.', job_id=job_id)
+    job = CrawlerJob.get_by_job(job_id)
+
+    sleep_time = current_app.config.get('CLI_HARVEST_SLEEP_TIME', 0.5)
+    sleep_counter = 0
+    try:
+        while job.status not in (JobStatus.ERROR, JobStatus.FINISHED):
+            if sleep_counter == 100:
```

Comment:
I don't see any statement that will break the loop in case you reach `sleep_counter` iterations. Maybe you want to put such condition in the `while` or in the next `if` statement.